marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/LoadFunctionMetaDataCommand.java

```diff
@@ -0,0 +1,76 @@
+package com.marklogic.hub.deploy.commands;
+
+import com.marklogic.appdeployer.command.AbstractCommand;
+import com.marklogic.appdeployer.command.CommandContext;
+import com.marklogic.appdeployer.command.SortOrderConstants;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.datamovement.ApplyTransformListener;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.QueryBatcher;
+import com.marklogic.client.document.ServerTransform;
+import com.marklogic.client.query.StructuredQueryBuilder;
+import com.marklogic.hub.DatabaseKind;
+import com.marklogic.hub.HubConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class LoadFunctionMetaDataCommand extends AbstractCommand {
+    @Autowired
+    private HubConfig hubConfig;
+    private boolean forceLoad = true;
+
+    public LoadFunctionMetaDataCommand() {
+        super();
+        // This command has to be executed after modules are loaded
+        setExecuteSortOrder(SortOrderConstants.LOAD_MODULES + 1);
+    }
+    public void setForceLoad(boolean forceLoad) {
+        this.forceLoad = forceLoad;
+    }
+
+
+    @Override
+    public void execute(CommandContext context) {
+        DatabaseClient databaseClient = hubConfig.newStagingClient(hubConfig.getDbName(DatabaseKind.MODULES));
+        DataMovementManager dataMovementManager = databaseClient.newDataMovementManager();
+
+        StructuredQueryBuilder sb = databaseClient.newQueryManager().newStructuredQueryBuilder();
+
+
+        ServerTransform serverTransform = new ServerTransform("ml:loadFunctionMetaData");
+        serverTransform.addParameter("forceLoad", String.valueOf(forceLoad));
+
+        ApplyTransformListener transformListener = new ApplyTransformListener()
+            .withTransform(serverTransform)
+            .withApplyResult(ApplyTransformListener.ApplyResult.IGNORE)
+            .onBatchFailure((batch, throwable) -> {
+                for(String item: batch.getItems()) {
+                    logger.error("Unable to load function metadata for: " + item);
+                }
+                logger.error(throwable.getMessage());
+                throwable.printStackTrace();
+            });
+
+        /*
+        Query for uris "/data-hub/5/mapping-functions/" and "/datahub/ext/mapping/" which are reserved for mapping functions
```

Comment:
Would a user place their custom mapping functions under "/datahub/ext/mapping"? DHF is generating custom modules under "/custom-modules" - should mappings go under "/custom-modules" as well? I think these locations won't be perceived as consistent by developers. "/datahub/" can also be easily confused with "/data-hub/", which is where product modules go.

Reply:
@ryanjdew ,
Based on the https://wiki.marklogic.com/pages/viewpage.action?spaceKey=ENGINEERING&title=Declarative+Mapping+Plan, currently
1. OOTB functions go to "/data-hub/5/mapping-functions/*.sjs|mjs|xqy"
2. Custom functions "/datahub/ext/mapping/*.sjs|mjs|xqy"

Are you ok with changing the custom functions to "/custom-modules/mapping-functions/*.sjs|mjs|xqy ?

Reply:
@srinathgit I'm fine with that and think it is worth changing for consistency.